### ISSUE TITLE
Cant generate release archive for iOS 'device_info/DeviceInfoPlugin.h' file not found

### ISSUE THREAD

**Issue by MiguelCatalan** [flutter/flutter#20564]
I keep getting `'device_info/DeviceInfoPlugin.h' file not found` error each time I want to generate a relase for iOS.

## Steps to Reproduce

**At IntelliJ:**
- `flutter upgrade`
- `flutter clean`
- `cd ios` and `pod install` (just in case)
- `flutter build ios --release -t lib/main_production.dart` (I use different main files for each environment, not related with this error at all)

**At xcode:**
- Set ***Product -> Scheme -> Runner***
- Set ***Product -> Destination -> Generic iOS Device***
- Generate build with ***Product -> Archive***

And this is what I get every time:
![image](https://user-images.githubusercontent.com/3306324/44078689-727be3b6-9fa7-11e8-83c3-d864ac201cf4.png)

Already try to change the channel to and from:
- `Beta`
- `Dev`
- `Master`

Already done `flutter upgrade`, `flutter clean`, `pod install` several times in each step of the process.

## Logs

The command `flutter doctor -v` returns: 
```
[✓] Flutter (Channel master, v0.5.9-pre.30, on Mac OS X 10.13.6 17G2208, locale en-ES)
    • Flutter version 0.5.9-pre.30 at /Users/miguelcatalan/Projects/Flutter/flutter
    • Framework revision 33c4cd0f39 (71 minutes ago), 2018-08-13 23:43:49 -0700
    • Engine revision e54bc4ea18
    • Dart version 2.0.0-dev.69.5.flutter-eab492385c

[✓] Android toolchain - develop for Android devices (Android SDK 27.0.3)
    • Android SDK at /Users/miguelcatalan/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-27, build-tools 27.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-915-b08)
    • All Android licenses accepted.

[✓] iOS toolchain - develop for iOS devices (Xcode 9.4.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 9.4.1, Build version 9F2000
    • ios-deploy 1.9.2
    • CocoaPods version 1.5.2

[✓] Android Studio (version 3.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-915-b08)

[!] IntelliJ IDEA Ultimate Edition (version 2018.2.1)
    • IntelliJ at /Applications/IntelliJ IDEA.app
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
    • For information about installing plugins, see
      https://flutter.io/intellij-setup/#installing-the-plugins

[✓] IntelliJ IDEA Community Edition (version 2017.3.5)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 26.0.1
    • Dart plugin version 173.4700

[✓] Connected devices (1 available)
    • Athena • 9285f26e88af3ac8461ad57ac5ae18fbfacd95cf • ios • iOS 11.4.1

! Doctor found issues in 1 category.

```

**Comment by MiguelCatalan** [comment]
Guys, my company is not able to ship any release because of this. We are missing a big business opportunity. Could anyone help, please?

**Comment by zoechi** [comment]
Can you please post the output of `pod --version`?
You could try https://github.com/flutter/flutter/issues/17943#issuecomment-392329305 or https://github.com/flutter/flutter/issues/17943#issuecomment-392911916

**Comment by MiguelCatalan** [comment]
@zoechi the output of `pod --version` is: **1.5.2**

**Comment by MiguelCatalan** [comment]
Upgraded to **1.5.3**. No improvements

**Comment by MiguelCatalan** [comment]
Removed brew cocoapods and stuck with the gem. No improvements

**Comment by MiguelCatalan** [comment]
Any more ideas @zoechi ?

**Comment by MiguelCatalan** [comment]
Created a new related issue at #21073

**Comment by raghaaavvvv** [comment]
I am facing the same issue while archiving the ios for production

**Comment by muhsinzyne** [comment]
I am too facing the same issue when I am taking build at my real device. 

the same item is working with simulator perfectly. any helps so far ?

**Comment by alsedev** [comment]
Same issue.  Any solution? Workaround?

**Comment by iamatulkumar** [comment]
Go to iOS directory and run pod clean and pod install,
Or open xcode and add this in framework

**Comment by alsedev** [comment]
pod clean
[!] Unknown command: `clean`
I should say: I am using Flutter channel master version 1.19.0-2.0.pre.143 & Dart 2.9.0-10.0.dev

**Comment by ionda88** [comment]
same issue

**Comment by netgfx** [comment]
Same issue when trying to run a release version `Module 'device_info' not found` did flutter clean and `pod cache clean --all` no changes. I'm on latest dev 1.23

**Comment by ayham95** [comment]
Just a +1 to check..
Make sure you're building for the correct destination:
in **XCode**
Product -> Destination -> Generic iOS Device
Then archive..

**Comment by aligurelli** [comment]
i am facing with same error....any solution?

**Comment by muhsinzyne** [comment]
@aligurelli  Please use Xcode through Xcode project.workspace instead of Xcode project.project

**Comment by heilmela** [comment]
> @aligurelli Please use Xcode through Xcode project.workspace instead of Xcode project.project

This solved the issue for me. Could your elaborate on this fact, I dont really get what's going on.

**Comment by edgarfroes** [comment]
Running `flutter build ios` before building in xcode using resolved the issue for me.

**Comment by surveshoeb** [comment]
Any update on this issue?

**Comment by razikallayi** [comment]
I may have found the answer from this comment:

https://github.com/flutter/flutter/issues/21989#issuecomment-513286777

Seems like by opening up the project by double clicking `Runner.xcworkspace` instead of `Runner.xcodeproj` xcode can finally figure out where everything is.  Even though you can still debug in an emulator and on a connected phone from the `.xcodeproj` file...

Seems strange that it would fix it.  But I'm happy to have an Archived build now.

_Originally posted by @Kenoshen in https://github.com/flutter/flutter/issues/43986#issuecomment-573171119_

**Comment by Masumi-M** [comment]
I had the same issue today and tried all the solution above, but it didn't work for me :(

However, I solved it out by the following steps.
1. I Made a repository in Github and pushed my project.
2. I cloned my repository from GitHub to another directory.
3. Rebuild the project by `flutter build ios`.

I guess there was some sort of error in the `build` folder or `ios/Pods` folder. 
I'm happy if this solution helps someone :)
